streams/writable-streams/start.js

```diff
@@ -93,15 +95,74 @@ promise_test(() => {
 }, 'underlying sink\'s write or close should not be invoked if the promise returned by start is rejected');
 
 promise_test(t => {
-  const rejection = { name: 'this is checked' };
   const ws = new WritableStream({
     start() {
       return {
-        then(onFulfilled, onRejected) { onRejected(rejection); }
+        then(onFulfilled, onRejected) { onRejected(error1); }
       };
     }
   });
-  return promise_rejects(t, rejection, ws.getWriter().closed, 'closed promise should be rejected');
+  return promise_rejects(t, error1, ws.getWriter().closed, 'closed promise should be rejected');
 }, 'returning a thenable from start() should work');
 
+promise_test(t => {
+  const ws = recordingWritableStream({
+    start(controller) {
+      controller.error();
+    }
+  });
+  return promise_rejects(t, new TypeError(), ws.getWriter().write('a'), 'write() should fail')
```

Comment:
The short answer appears to be "for consistency with the _closed_ and _closing_ states". There's no [[storedError]] for us to return in those states. It's been that way since @tyoshino added add getWriter() in https://github.com/whatwg/streams/pull/488. I haven't looked in enough depth to see whether it was explicitly discussed.

It wouldn't be difficult to special case the "errored" state [here](https://streams.spec.whatwg.org/#writable-stream-default-writer-write), but I don't find it compelling. Let's see what @tyoshino thinks.